iaf-commons/src/main/java/nl/nn/adapterframework/util/StringResolver.java

```diff
@@ -0,0 +1,509 @@
+/*
+   Copyright 2013, 2014 Nationale-Nederlanden, 2020 - 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+
+/**
+ * Provide functionality to resolve ${property.key} to the value of the property key, recursively.
+ *
+ * @author Johan Verrips
+ */
+public class StringResolver {
+	// Not allowed to use a static reference to the logger in this class.
+	// Log4j2 uses StringResolver during instantiation.
+
+	private static final String VALUE_SEPARATOR=":-";
+
+	public static final String DELIM_START = "${";
+	public static final String DELIM_STOP = "}";
+
+	private static Collection<AdditionalStringResolver> additionalStringResolvers = null;
+
+	/**
+	 * Look up the key in the environment with {@link System#getenv(String)} and next in the System properties with
+	 * {@link System#getProperty(String, String)}. The {@link SecurityException} if thrown, is hidden.
+	 *
+	 * @param key The key to search for.
+	 * @param def The default value to return, may be {@code null}.
+	 * @return the string value of the system property, or the default value if
+	 *         there is no property with that key.
+	 *         May return {@code null} if the default value was {@code null}.
+	 *
+	 * @since 1.1
+	 */
+	public static String getSystemProperty(String key, String def) {
+		try {
+			String result = System.getenv().get(key);
+			if (result!=null) {
+				return result;
+			}
```

Comment:
Dit kan /moet eigenlijk naar `Environment`

Reply:
Is een Public method maar wordt eigenlijk alleen intern gebruikt in de `StringResolver`.

Maar je hebt gelijk dat het eigenlijk beter naar `Environment` kan.